src/raftstore/store/peer.rs

```diff
@@ -404,6 +415,40 @@ impl Peer {
         down_peers
     }
 
+    pub fn check_stale_state(&mut self, d: Duration) -> StaleState {
+        // update leader_missing_time according to current state
+        if self.leader_id() == raft::INVALID_ID {
+            if self.leader_missing_time.is_none() {
+                self.leader_missing_time = Some(Instant::now())
+            }
+        } else if self.is_initialized() {
+            // When a peer is not initialized, it has no data at storage.
+            // Even if a leader sends heartbeats to it, we consider it as
+            // in the `leader missing` state. That is because if it's isolated from the leader
+            // before it could successfully receive snapshot from the leader and
+            // apply that snapshot, no raft ready event will be triggered,
+            // so that we could not detect the leader is missing for it at here.
+            self.leader_missing_time = None
+        }
+
+        // check whether current peer is stale
+        let duration = match self.leader_missing_time {
+            Some(t) => t.elapsed(),
+            None => Duration::new(0, 0),
+        };
+        if duration >= d {
+            // reset the leader missing time to avoid sending the same tasks to
+            // PD worker continuously on leader missing timeout
```

Comment:
+            // Resets the leader_missing_time to avoid sending the same tasks to the
 +            // PD worker continuously during the leader missing timeout.